File: src/Samples/Net_Framework/MVVM/Movie.cs
using CommunityToolkit.Mvvm.ComponentModel;
using System;
using System.ComponentModel.DataAnnotations;

namespace OpenSilver.Samples.Showcase;

public partial class Movie : ObservableValidator
{
    [ObservableProperty]
    [Required(ErrorMessage = "Title is required.")]
    private string _title;

    [ObservableProperty]
    private MovieStatus _status;

    [ObservableProperty]
    private DateTime? _dateLastWatched;

    [ObservableProperty]
    private bool _favorite;

    [ObservableProperty]
    [Range(0.0, 1.0, ErrorMessage = "Rating must be between 0 and 1.")]
    private double _myRating;
}

public enum MovieStatus
{
    ToWatch,
    Watching,
    Watched,
    ToWatchAgain,
    Abandoned
}

File: src/Samples/Net_Framework/MVVM/Movie.vb
Imports System.ComponentModel.DataAnnotations
Imports CommunityToolkit.Mvvm.ComponentModel

Namespace OpenSilver.Samples.Showcase

    Partial Public Class Movie
        Inherits ObservableValidator

        <ObservableProperty>
        <Required(ErrorMessage:="Title is required.")>
        Private _title As String

        <ObservableProperty>
        Private _status As MovieStatus

        <ObservableProperty>
        Private _dateLastWatched As DateTime?

        <ObservableProperty>
        Private _favorite As Boolean

        <ObservableProperty>
        <Range(0.0, 1.0, ErrorMessage:="Rating must be between 0 and 1.")>
        Private _myRating As Double
    End Class

    Public Enum MovieStatus
        ToWatch
        Watching
        Watched
        ToWatchAgain
        Abandoned
    End Enum

End Namespace
